tests/compiler/class.optimized.wat

```diff
@@ -131,18 +131,6 @@
   i32.const 5
   call $~lib/rt/stub/__alloc
   local.tee $0
-  i32.const 0
-  i32.store
-  local.get $0
-  i32.const 0
-  i32.store offset=4
-  local.get $0
-  i32.const 0
-  i32.store offset=8
-  local.get $0
-  i32.const 0
-  i32.store offset=12
-  local.get $0
```

Comment:
It looks pretty suspicious!

Reply:
Looks like that's an inlined `Array` constructor setting `.buffer`, `.dataStart`, `.byteLength` and `.length`, which all are being assigned in the constructor itself. Might be expected if the allocation knows that it does not have to zero-initialize these. Is it?

Reply:
Indeed, that's what's happening here, following this reasoning I did a slight change and things seem to be working again, but I could use some clarification from your end.

It seems that even though fields are not definitely initialized at the class level (when calling `makeFieldInitializationInConstructor`) the class expects them to be initialized with their default values.  So in https://github.com/AssemblyScript/assemblyscript/pull/1155/commits/8c6a55440226d0d2364f12d7142a33c9c3965dd5, I don't skip the default init anymore but I don't mark the property as fully initialized unless it complies with the strict property conditions. In my first approach, I was under the impression that skipping the actual default initialization if the field is not compliant was the right thing to do.

Reply:
> It seems that even though fields are not definitely initialized at the class level (when calling makeFieldInitializationInConstructor) the class expects them to be initialized with their default values

I'm assuming this makes sense / is required from an allocation perspective and that's why I was hitting the out of range error earlier.